### PR TITLE
[NEW] Added GET/POST channels.notifications

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Added two endpoints: GET/POST channels.notifications, that returns and updates the notification settings from a specific channel. Requested in [this](https://github.com/RocketChat/Rocket.Chat.iOS/issues/288) issue.

### CODE REVIEW

**Comment by rafaelks** [comment]
@MarcosSpessatto Amazing! Do you have the docs already?

**Comment by MarcosSpessatto** [comment]
@rafaelks, [docs](https://github.com/RocketChat/docs/pull/637).

File: packages/rocketchat-push-notifications/server/methods/saveNotificationSettings.js [hunk 1]
```diff
@@ -1,49 +1,59 @@
...
+			'desktopNotificationDuration': {
+				updateMethod: (subscription, value) => RocketChat.models.Subscriptions.updateDesktopNotificationDurationById(subscription._id, value)
+			},
+			'audioNotificationValue': {
+				updateMethod: (subscription, value) => RocketChat.models.Subscriptions.updateAudioNotificationValueById(subscription._id, value)
+			}
+		};
+		const isInvalidNotification = !Object.keys(notifications).includes(field);
+		const basicValuesForNotifications = ['all', 'mentions', 'nothing', 'default'];
+		const fieldsMustHaveBasicValues = ['emailNotifications', 'audioNotifications', 'mobilePushNotifications'];
```
Comment (rodrigok) [inline packages/rocketchat-push-notifications/server/methods/saveNotificationSettings.js]: Is this missing `desktopNotifications`?

File: tests/end-to-end/api/02-channels.js [hunk 2]
```diff
@@ -559,4 +551,62 @@ describe('[Channels]', function() {
...
+				}
+			})
+			.expect('Content-Type', 'application/json')
+			.expect(200)
+			.expect((res) => {
+				expect(res.body).to.have.property('success', true);
+			})
+			.end(done);
+
+		it('/channels.leave', async(done) => {
```
Comment (rodrigok) [inline tests/end-to-end/api/02-channels.js]: Why this `it` is inside the `it('POST /channels.notifications', (done) => {`?

File: packages/rocketchat-api/server/v1/channels.js [hunk 3]
```diff
@@ -798,23 +798,16 @@ RocketChat.API.v1.addRoute('channels.unarchive', { authRequired: true }, {
...
-			return subscription;
-		};
 		const { roomId } = this.requestParams();
 
 		if (!roomId) {
 			return RocketChat.API.v1.failure('The \'roomId\' param is required');
 		}
 
-		const subscription = RocketChat.models.Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId);
+		const subscription = RocketChat.models.Subscriptions.findOne({}, {fields: {_room: 0, _user: 0, $loki: 0}});
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/channels.js]: Wrong find, please revert to the old one

**Pull request merged**